src/main/java/com/icthh/xm/tmf/ms/communication/messaging/AbstractDeliveryReportListener.java

```diff
@@ -58,22 +62,24 @@ protected DeliveryReceipt getShortMessage(DeliverSm deliverSm) {
         return null;
     }
 
-    protected String getMessageId(DeliveryReceipt deliveryReceipt){
-        String messageId = deliveryReceipt.getId();
+    protected String getMessageId(DeliveryReceipt deliveryReceipt) {
```

Comment:
@oparashchyn please write a test for your changes. 
thank you

Reply:
I have added unit tests to check that message id is taken from either delivery receipt or optional parameters.
IMHO, it is not very common practice to test private/protected methods.
It would be very nice to have working sonar coverage report, I bet these lines of code are already covered by other tests.

Reply:
IMHO the idea is not to make segregation between private / protected / public methods during covering logic by the test but rather validate the influence that was made by each particular MR. For example, previously we would have and NPE when deliveryReceipt was null and you have fixed it. And I believe that should be reflected in the test.

Reply:
@amedvedchuk But better to cover this logic over public method, that used it. Is not it? @oparashchyn Does this method cover by test from the public method?

Reply:
@ishkurko for me it is not the question which method to cover. You can choose the best fit option in each case. 
There was an MR which actually change the behavior but without _any_ test that highlights that the behavior was changed.

Reply:
for sure that usually you use public methods as an entry point to invoke from the tests, and they also involve private and others methods to produce the results which then verified by the test.

Reply:
@ishkurko I have added tests the cover this method. As I said previously, I have added unit tests to check that message id is taken from either delivery receipt or optional parameters, these tests cover both cases when _deliveryReceipt_ is null and not null.
@amedvedchuk Please take a look.